partcad/src/partcad/converter.py

```
@@ -0,0 +1,169 @@
+        mesh = BRepMesh_IncrementalMesh(shape, linear_deflection, True, angular_deflection, True)
+        mesh.Perform()
+
+        writer = StlAPI_Writer()
+        if not writer.Write(shape, stl_file):
+            raise IOError(f"Failed to write STL file: {stl_file}")
+
+    @staticmethod
+    def _read_brep(brep_file: str) -> TopoDS_Shape:
+        """Read a BREP file and return its shape."""
+        shape = TopoDS_Shape()
+        with open(brep_file, "r") as file:
```

Ruff (0.8.2)

Unnecessary open mode parameters

Remove open mode parameters

(UP015)